### ISSUE TITLE
Change order of containers initialization

### ISSUE THREAD

**Issue by igagis** [actions/runner#816]
**Describe the enhancement**
Change the order in which job services and job container are started.
Currently the order is as follows:
1. start job container
1. start all job services
1. wait until all job services are ready

The proposal is to change the order to the following:
1. start all job services
1. wait until all job services are ready
1. start job container

it makes more sense to start job container after all the services are started, as the job container might use services already early stage.

The actual need for this change arose from this request https://github.com/actions/virtual-environments/issues/2095
Long story short: it is desired to use services to initialize `qemu` and `binfmt-support` before the actual job container is started.

**Comment by igagis** [comment]
@umarcor FYI

**Comment by mpmkp2020** [comment]
I am hitting the similar issue, is anyone working on it? What is the milestone for it?
